### PR TITLE
Refactor heisskleber core, remove synchronous implementations

### LINKED ISSUE #130: Remove config options that are not heisskleber related
Heisskleber was mostly developed with specific applications in mind and still has some baggage associated with that. 

To clean things up, obscure references in configuration files should go. 

e.g.: "mapping" and "source_id" in MqttConf.

### LINKED ISSUE #132: Async Serial connections with aioserial

### PR DESCRIPTION
**Pull request by pacjac**
Significant refactor of our code base, breaking most things along the way.  

Key changes include:

## Code

- Implemented new packer/unpacker interface for more generic usage, being less opinionated about what type of data format to use (I still like dictionaries, but someone else might prefer something with type checking, data classes or lists)
- Improved type hinting and generics usage
- Refactored to fully async implementation, removing synchronous classes
- Added async wrapper for pyserial
- Enhanced AsyncTcpSource with retry and reconnection capabilities
- Renamed various components for consistency

## Repo
- Switched from poetry to uv. We love uv!
- Added nox support for development workflows
- Updated documentation with myst-parser and sphinx

The aim is to improve code maintainability, performance, and developer experience.

## Resolved issues:
- #133, new configuration management. Allows loading from file (as requested in #95) (yaml and json).
- Cleans up configurations, fixes #130.
- Introduces the work of @a-w, fixing #129 and #101. 
- A general packer / unpacker refactor fixes for #93. 
- Documentation improvements for #3 and #105. 
- Adds async serial connections, closes #132.

### CODE REVIEW

**Review by k323r — CHANGES_REQUESTED** [review]
Great work! I have a few, small nitpicks here and there, no showstopper, though. Great progress towards v1!

Please review my comments, discard them as you see fit (or incorporate them). Only test_05 from the tcp test suite seems to be broken and I also don't understand it. :shrug:

File: src/heisskleber/__init__.py
[the remark this answers is not in the document]
Comment (pacjac) [reply src/heisskleber/__init__.py:1]: Added.

File: src/heisskleber/console/sink.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        verbose: bool = False,
+        packer: Packer[T] = json_packer,  # type: ignore[assignment]
+    ) -> None:
+        self.verbose = verbose
+        self.pretty = pretty
+        self.packer = packer
+
+    async def send(self, data: T, topic: str | None = None, **kwargs: dict[str, Any]) -> None:
+        """Serialize data and write to console output."""
+        verbose_topic = topic + ":\t" if topic else ""
```
[the remark this answers is not in the document]
Comment (pacjac) [reply src/heisskleber/console/sink.py:23]: Changed accordingly.

File: src/heisskleber/console/source.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class ConsoleSource(AsyncSource[T]):
+    """Read stdin from console and create data of type T."""
+
+    def __init__(
+        self,
+        unpacker: Unpacker[T] = json_unpacker,  # type: ignore[assignment]
+    ) -> None:
+        self.queue: asyncio.Queue[tuple[T, dict[str, Any]]] = asyncio.Queue(maxsize=10)
```
[the remark this answers is not in the document]
Comment (pacjac) [reply src/heisskleber/console/source.py:17]: Typing is dependend on the packer. The ConsoleSource is of type AsyncSource[T]. If I pass it an unpacker of Type dict[str, Any] (which means the unpacker returns a payload of type dict (and an additional meta-dict of type dict)), then the type of the ConsoleSource is [dict[str, Any]].

There is a document on writing your own custom packers and unpackers.

File: src/heisskleber/mqtt/sink.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import aiomqtt
+
+from heisskleber.core import AsyncSink, Packer, json_packer
+from heisskleber.core.utils import retry
+
+from .config import MqttConf
+
+T = TypeVar("T")
+
+logger = logging.getLogger(__name__)
```
[the remark this answers is not in the document]
Comment (pacjac) [reply src/heisskleber/mqtt/sink.py:17]: Done.

**Review by k323r — APPROVED** [review]

**Pull request merged**